### PR TITLE
Fix get request with query

### LINKED ISSUE #134: Rest client. Get request with query fails if the limit is not specified
# Issue summary
With this kind of request
```
client.get({
  path: 'redirects',
  query: {
	  "path": 'some_path',
  },
});
```

I get error, because limit is undefined

## Expected behavior

Successful request to the api

## Actual behavior

I get error UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'toString' of undefined
which comes from this row https://github.com/Shopify/shopify-node-api/blob/7c9201af556c6f82deb936215f5099d08e35f727/src/clients/rest/rest_client.ts#L35

---

## Checklist

- [x] I have described this issue in a way that is actionable (if possible)

### PR DESCRIPTION
**Pull request by derom**
### WHY are these changes introduced?

Fixes #134

### WHAT is this pull request doing?

Added test to catch the described error. Made limit optional, because it has default value (50)

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above
- [x] I have added/updated tests for this change
- [ ] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

**Review by thecodepixi — APPROVED** [review]
This makes sense to me. We should definitely be able to make REST requests without passing a limit, since the API will just apply its default limit to the request. Thanks for catching that.

**Comment by derom** [comment]
Thanks for review @thecodepixi
I've pushed some changes. ```limit``` in PageInfo is non-optional again. I realized that the limit is always known and we should just set the default value if a user didn't specify the limit. Also, unused type PageInfoParams was removed

File: src/clients/rest/rest_client.ts [hunk 1]
```diff
@@ -31,8 +31,14 @@ class RestClient extends HttpClient {
 
     const link = ret.headers.get('link');
     if (params.query && link !== undefined) {
+      const defaultLimit = '50';
```
Comment (thecodepixi) [inline src/clients/rest/rest_client.ts:34]: I might put this as a static variable on the class instead, as we have with the regex: 

```suggestion
     private static DEFAULT_LIMIT = '50';
```

and then we can still reference it with the same type of conditional statement.

**Comment by thecodepixi** [comment]
Taking another look at this now, and I'm not sure we can entirely do away with `PageInfo, but have asked for additional opinions on that front.

**Comment by paulomarg** [comment]
I agree, it's a bit strange that it doesn't cause any tests to break, but we're definitely still using that type in the REST client class.

**Comment by derom** [comment]
@thecodepixi @paulomarg PageInfoParams was replaced by PageInfo in this commit https://github.com/Shopify/shopify-node-api/commit/53b5e0779e55b9e554aa88b1a8d0d34c4e655111
I didn't find any usages of that interface https://github.com/Shopify/shopify-node-api/search?q=PageInfoParams

File: src/clients/rest/rest_client.ts (hunk 1 above)
Comment (derom) [reply src/clients/rest/rest_client.ts]: +

**Comment by thecodepixi** [comment]
Ah OK, maybe that was my misunderstanding. You're proposing just removing `PageInfoParams` but not `PageInfo`?

**Comment by derom** [comment]
Yes, only PageInfoParams

File: src/clients/rest/rest_client.ts [hunk 2]
```diff
@@ -31,8 +32,13 @@ class RestClient extends HttpClient {
 
     const link = ret.headers.get('link');
     if (params.query && link !== undefined) {
+      let limit = RestClient.DEFAULT_LIMIT;
```
Comment (thecodepixi) [inline src/clients/rest/rest_client.ts:35]: Nit: This feels a bit redundant. Can we accomplish this without the reassignment on this line?

**Comment by thecodepixi** [comment]
> Yes, only PageInfoParams

Oh then yeah, if that's no longer needed I would think it's fine to remove.

File: src/clients/rest/rest_client.ts (hunk 2 above)
Comment (derom) [reply src/clients/rest/rest_client.ts]: Agree. I think it looks better now

**Review by paulomarg — APPROVED** [review]
I like it too! 🚀

**Comment by derom** [comment]
Should I do anything else to merge this pr?

**Comment by thecodepixi** [comment]
> Should I do anything else to merge this pr?

I gotchu!

**Pull request merged**